FrontEnd/src/App.js

```diff
@@ -0,0 +1,155 @@
+import React, { Component } from 'react';
+import { render } from 'react-dom';
+import './index.css';
+import Recaptcha from 'react-recaptcha';
+
+export default class App extends Component {
+  render() {
+    return (
+      <Register />
+    );
+  }
+}
+
+// eslint-disable-next-line
+const validEmailRegex = RegExp(/^(([^<>()\[\]\.,;:\s@\"]+(\.[^<>()\[\]\.,;:\s@\"]+)*)|(\".+\"))@(([^<>()[\]\.,;:\s@\"]+\.)+[^<>()[\]\.,;:\s@\"]{2,})$/i);
+
+const validateForm = (errors) => {
+    let valid = true;
+    Object.values(errors).forEach(
+      (val) => val.length > 0 && (valid = false)
+    );    
+    return valid;
+}
+
+class Register extends Component {
+  constructor(props) {
+    super(props);
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.recaptchaLoaded = this.recaptchaLoaded.bind(this);
+    this.verifyCallback = this.verifyCallback.bind(this);
+    
+    this.state = {
+      firstName: null,
+      email: null,
+      message: null,
+      isVerified: false,
+      errors: {
+        firstName: '',
+        lastName: '',
+        email: '',
+        message: ''
+      }
+    };
+  }
+
+  handleChange = (event) => {
+    event.preventDefault();
+    const { name, value } = event.target;
+    let errors = this.state.errors;
+
+    switch (name) {
+      case 'firstName': 
+        errors.firstName = value.length == 0 ? 'Please enter your first name' : '';
```

Comment:
In React, you don't want to be directly mutating your state, because the component won't re-render until other props change or another setState is called.

Could you refactor this to not directly mutate your state while keeping the same effect?

Reply:
I'm not following.  Please show me a better approach.

Reply:
This article does a better job of describing it than I would:
https://www.freecodecamp.org/news/handling-state-in-react-four-immutable-approaches-to-consider-d1f5c00249d5/